sqlglot/dialects/singlestore.py

```diff
@@ -79,6 +99,21 @@ class Parser(MySQL.Parser):
                 this=this,
                 to=to,
             ),
+            TokenType.DCOLON: lambda self, this, path: build_json_extract_path(exp.JSONExtract)(
+                [this, exp.Literal.string(path.this.this)]
+            )
+            if isinstance(path, exp.Column)
+            else build_json_extract_path(exp.JSONExtract)([this, path]),
```

Comment:
When can `path` _not_ be a `Column` here? Seems like singlestore only expects `foo::bar` based on the docs. Ditto for `DCOLONDOLLAR` and `DCOLONPERCENT`.

Reply:
That's true. According to SingleStore documentation, in correct queries path should always be a column

Reply:
Good, we can then assume this happy path and simplify the logic here.

Reply:
You should do `path.name` instead.

Reply:
Wow. This worked. Thanks for the suggestion.